src/tests/mm/MmAllocateSystemMemory.c

```diff
@@ -1,8 +1,58 @@
 #include <xboxkrnl/xboxkrnl.h>
 
 #include "util/output.h"
+#include "assertions/defines.h"
+
+#define SYSTEM_MEMORY_BASE    0xD0000000
+#define SYSTEM_MEMORY_SIZE    (512 * 1024 * 1024) // = 0x20000000
+#define SYSTEM_MEMORY_END     (SYSTEM_MEMORY_BASE + SYSTEM_MEMORY_SIZE - 1) // 0xEFFFFFFF
+#define IS_SYSTEM_ADDRESS(Va) (((ULONG)(Va) - SYSTEM_MEMORY_BASE) <= (SYSTEM_MEMORY_END - SYSTEM_MEMORY_BASE))
+
+#define CHECK_ALIGNMENT(size, alignment) (((size) % (alignment)) == 0)
 
 void test_MmAllocateSystemMemory()
 {
-    /* FIXME: This is a stub! implement this function! */
+    const char* func_num = "0x00A7";
+    const char* func_name = "MmAllocateSystemMemory";
+    BOOL test_passed = 1;
+    print_test_header(func_num, func_name);
+
+   // NOTE: these tests will be marked as failed if not enough memory can be allocated for them
```

Comment:
```suggestion
   // NOTE: these tests could be marked as failed if not enough memory can be allocated for them
```

Reply:
For clarification, if memory allocated failed. The following reasons could be the cause:
- Not enough memory to allocated (previous xbe used up all the memory and didn't freed them up)
- defective chipset (hardware issue)
- stub function which only return null pointer. (custom kernel issue, not an issue with the test itself)
- etc